### DISCUSSION TITLE
Choosing an IronOS compatible device

### DISCUSSION THREAD

**Discussion by debo** [Ralim/IronOS#1759] [Q&A] [answered]
Hello everyone, first of all please let me thank @Ralim and the IronOS community for the impressive work you are doing.

Now, to the question; I am looking for a portable soldering iron to use for minor soldering activities, more specifically to build mechanical keyboards, ideally to the extent of building something like a Corne or a Discipline keyboard entirely from scratch.

I was looking to the usual suspects, TS80P, TS100, TS101 and Pinecil V2, but I'm not quite sure which one would suit me best and which one is admittedly the best choice.

Which one would you recommend or which one would you buy if it was your money leaving budget aside for a moment? I also welcome alternative recommendations if you believe there are better options out there.

One thing that I also think might be important for me is IronOS support. I feel like the opportunity to use a platform that is open source friendly would be a good plus but it's also not a deal breaker.

Thank you in advance for your help and advice.

**Comment by Ralim** [comment, marked as answer]
If it were me and budget were not object, I'd get a Pinecil V2 and a S60.
Okay, why two:
The PinecilV2 is a champ and can handle _ basically anything_ you throw at it. But its main annoyance is the tips are moderately large for tight spaces. Ergo the S60 for the thinner JBC tips.

If your not doing fine soldering or small SMD work, just the Pinecil (V2) would be my goto.

I would **not** recommend (TS100,TS80) as they are both entirely superceeded in terms of newer products.
TS101 is often more expensive than PinecilV2 and doesn't really give you much extra for the money. TS80P isn't bad, but I find the S60 a better value proposition usually.

**Reply by ia** [reply]
> The PinecilV2 is a champ and can handle _ basically anything_ you throw at it. But its main annoyance is **the tips are moderately large for tight spaces**.
> 
> If your not doing fine soldering or **small SMD work**, just the Pinecil (V2) would be my goto.

Since we're here... but have you ever tried [those custom short `T12` fine tips](https://pine64.com/product/pinecil-soldering-short-tip-set-fine/) with relatively thin tips on the edges from `PINE64` on the `PinecilV2`? And if yes - how smaller the edge tips in real life are in comparing with that limited set for `TS80/P` tips? I was curious myself how good & convenient those `T12` could be to use for SMD - at least up to the sizes of, let's say `0603`.

> Ergo the S60 for the thinner JBC tips.

And how you would describe the current state of support of `S60` in the current `dev`? ;)
_**Everything** runs smooth_ besides full support of scaling for larger OLED screens (just like with `TS101`, right?), is this correct?

As always, thanks.

**Reply by Ralim** [reply]
> Since we're here... but have you ever tried [those custom short `T12` fine tips](https://pine64.com/product/pinecil-soldering-short-tip-set-fine/) with relatively thin tips on the edges from `PINE64` on the `PinecilV2`? And if yes - how smaller the edge tips in real life are in comparing with that limited set for `TS80/P` tips? I was curious myself how good & convenient those `T12` could be to use for SMD - at least up to the sizes of, let's say `0603`.

I've mostly used them with `0603` and `0402` and have found them to work fine, the thinner knife one worked well for me. `0402` and `0201` are a bit better with the tiniest one but you cant do big thermal pads with that tip. I tend to stick to the 2nd smallest and call it done.

> 
> > Ergo the S60 for the thinner JBC tips.
> 
> And how you would describe the current state of support of `S60` in the current `dev`? ;) _**Everything** runs smooth_ besides full support of scaling for larger OLED screens (just like with `TS101`, right?), is this correct?

1 minor bug to fix for DC supplies (next 2-3 days) and otherwise just that the OLED currently only draws in top left 1/3rd until I finish all the OLED refactoring. (So its just cosmetic and 1000% fixable when I have time).
I found IronOS performs a _lot_ better on 12V supplies than the stock firmware. But YMMV.

**Reply by debo** [reply]
> If it were me and budget were not object, I'd get a Pinecil V2 and a S60.

Thanks a lot @Ralim for chipping in and for the suggestion and also apologies for tagging you in the first post but I had to thank you.

I completely understand the rationale behind the "Why two" but, for completeness of argument, I wonder: "why not just the S60 then?" What are its limitations to require the pinecil as well?

**Comment by ia** [comment]
> There is always that one person
> ... that ask this question, and it appears to be me. Sorry not sorry :)

Hello. I'm sorry too to bother you with that :D ... but could you, please, just rename the title of this discussion so other people who will be looking at the topics list from the main Discussion page could get an idea what this discussion is about? Something like "Choosing IronOS-compatible iron". :) Thanks.

As to your question... I would second @Ralim on getting more than one: one for heavy duty & "medium size" work, and another one (just in case to have it always ready in your toolbox) for tiny SMD-like jobs or very thin wires and small pads/places/etc. If it would be a few years ago (i.e., _before_ `PinecilV2`), I would stick myself to `TS100`/`TS80P` couple. But now since we have more options, and if I would have unlimited budget, then even just out of curiosity I would try `PinecilV2`/`S60` couple myself.

As the happy owner of `TS80P` (FYI: I don't have anything else _yet_ so nothing to compare with), I could add that my own main & probably only one problem is the tips: they are very custom (not like almost universal `T12`), very limited (only 5 for your choice)... and relatively expensive (for soldering tips). Oh, and from the point of long-term support [by IronOS] `TS80P` model has interesting situation: since it supports varieties of features & power sources, more code must be compiled in to the firmware, but since the model is relatively old it has only `64K` flash (against `128K` in `PinecilV2` if I remember correctly) so it's _theoretically_ possible that there will be the time (not _soon_ but just a warning) when `IronOS` with all of its features just physically won't be able to fit into flash memory of `TS80P`. Howevery, since @Ralim & the crew do everything possible for optimizations for different models, I hope this won't be a problem for the next couple of years at least.

**Reply by Ralim** [reply]
Yeah one of the biggest reasons for S60 over TS80(P) is the cost and availability of tips.
TS80P's flash restriction is a paiin, and likely at some point in the future we will see more features being "not available" on it.

**Reply by debo** [reply]
No bother at all @ia title and body adapted. It was a bit late when I created the question and words were scarse in my brain.

Thanks a lot for the recommendation and for pointing out the firmware issue.

**Reply by ia** [reply]
> No bother at all @ia title and body adapted. It was a bit late when I created the question and words were scarse in my brain.

No-no, that's ok! Just thought that the more topic will be clear, the better for the forum community. :)

> Thanks a lot for the recommendation and for pointing out the firmware issue.

Just a tiny update to my own comment since now when I look at, it looks too negative towards `TS80P`. As I said, I'm more than happy in general but didn't highlight the actual positive sides (so fixing it now) which is - **super comfortable physical design** (that's subjective but nevertheless):
- the unit doesn't overheat much even when you use it relatively long with relatively high temperatures;
- the tips are relatively short (i.e., the distance between the edge of the iron & the edge of the tip) which is always a plus so the iron feels just like a very comfortable pen in your hand;
- hot swapping (literally) of the tips with bare hands is just a killer feature for not so experienced soldering hobbyists - when due to lack of experience you're not sure which tip will fit better[^1] for the current task while you have a long soldering session, you eventually ends up switching the tips back & forth for different tasks/surfaces/pads/elements (especially if you have only one iron)... so those holes in the conic thing on the tips which you may notice are not [only] for the show - they disperse heat really good so tips can be removed without a [serious] risk of burning fingertips (that part of the tips usually is not getting too hot at all!).

Anyway, whatever you will be using - happy soldering!

[^1]: or you just don't know _yet_ how to properly use only one kind of a tip for any job due to lack of experience as well

**Comment by debo** [comment]
> No-no, that's ok! Just thought that the more topic will be clear, the better for the forum community. :)

I know, and I am 100% in agreement, that's why I was more than happy to change it; it was a very good shout.

> Just a tiny update to my own comment since now when I look at, it looks too negative towards `TS80P`
> ...
> Anyway, whatever you will be using - happy soldering!

Got it and I have to admit that I do like the TS80P a lot I like the round comfortable shape and the release mechanism seems very well thought. I was leaning a lot towards buying that one at one point but I eventually pulled the trigger on the Pinecil because I believe serving the thriving community behind it is better. Now I can't wait for it to arrive.
